prediction/src/algorithms/segment/src/levelset_segment.py

```diff
@@ -0,0 +1,52 @@
+import SimpleITK as sitk
+
+
+def segment_nodule(image: sitk.Image, nodule: dict, lower_threshold=-600, upper_threshold=500):
+    """Args:
+      image: sitk.Image: Input Imae
+      nodule: dict: Nodule coordinates
+      lower_threshold:  Lower Hu threshold of the nodule to be segmented. Should be about the Hu of lung
+                      (Default value = -600)
+      upper_threshold: Upper Hu threshold of the nodule to be segmented. Should be less than the Hu of bone
+                       (Default value = 500)
+
+    Returns:
+        maskÆ filepath to segmentation mask
+        diameters: list of diameters
+        volumes: list of  volumes
+
+    """
+
+    seed = (nodule["x"], nodule["y"], nodule["z"])
+    # Create initial levelset based on the seed and the signed Maurer distance
+    seg = sitk.Image(image.GetSize(), sitk.sitkUInt8)
+    seg.CopyInformation(image)
+    seg[seed] = 1
+    seg = sitk.BinaryDilate(seg, 3)
+    init_ls = sitk.SignedMaurerDistanceMap(seg, insideIsPositive=True, useImageSpacing=True)
+
+    lsFilter = sitk.ThresholdSegmentationLevelSetImageFilter()
+    # Sets the Hu value range for the nodule
+    lsFilter.SetLowerThreshold(lower_threshold)
+    lsFilter.SetUpperThreshold(upper_threshold)
+    # Sets the error tolerance and number of iterations for the levelset solver
+    lsFilter.SetMaximumRMSError(0.02)
+    lsFilter.SetNumberOfIterations(1000)
```

Comment:
I think someone previously mentioned here in review about explaining or making these constants/or at least documenting the magic numbers a bit? See also ``BinaryDilat(seg, 3)``. Three!

Reply:
@lamby, I am a little unsure what you are asking.
The algorithm is iterative, and RMSError and number of iterations are the stopping criteria. None of the parameters are "special", and it seems a bit weird to have a line like
```
maximumRMSError = 0.02
lsFilter.SetMaximumRMSError(maximumRMSError)
```
Most of the parameters were taken from the SITK tutorial on segmentation, but some (curvature and thresholds) were changed to work better for lung segmentation. 
The parameters are either very easy to understand (the thresholds), or quite mathematically involved (curvature).  I've tried to add a few more comments, but as I said, I'm  not entirely sure what you're looking for.

Reply:
In this case it would be upper case for constants.... so `MAX_RMS_ERROR` perhaps. It may look weird at first, but it's usually best to declare constants at the top of the function or file.

Reply:
:+1:

Reply:
Hmm... not quite sure it increases readability in this case. After more consideration, I've moved the parameters to the args of the function, with reasonable defaults.